src/04-Fundamentals-2.mdx

```diff
@@ -1,4 +1,121 @@
++ Use the BEM way to make modifiers for the component (see reference)
+
+### Resources
++ [SASS's own reference](https://sass-lang.com/guide) or [CSS-Tricks](https://css-tricks.com/snippets/sass/)
++ [BEM's own reference](http://getbem.com/naming/) or [CSS-Tricks](https://css-tricks.com/bem-101/)
+
+## JavaScript
+
+'Finally', is what you're thinking to yourself probably. Yes, now we're getting into the good stuff. Up until now, everything you have coded is more or less 'static'. Well, that changes now. In this section, you will get to Javascript, it's newest version ECMAScript2015 (also called ES6) and, to a lesser extent, jQuery. We here at VI try to write our code in ES6 whenever possible. jQuery, while it used to be the standard and provides a lot of functionality that makes our lives easier, is being used less and less and it's more than likely that it will completely die eventually. That is why coding in 'plain' JS is always preferred.
+
+### Challenge
+
+Add the following functionality to the component, that you built in the section 'SCSS': When the button is clicked, a textbox underneath the already existing elements slides open. 
+
+<img style={{width: '100%'}} src={screenOpen}/>  
+
+### Hints and guidelines
+
++ Try to apply ES6 code when possible (it's the future!), but also try out jQuery here and there
++ To debug, use the tools that you are provided with (*hint* Chrome dev tools *hint*)
++ There's more than one way to skin a cat.. uhhh animate. CSS, jQuery and Javascript can all do this, ask your buddy for help!
++ ...
+
+#### TODO: Fill the list
+
+## Events
+
+Components need to communicate with each other. For example, you would probably want an open tooltip to close when the user opens the navigation on your page. In order for this to work, the navigation needs to tell the tooltip to disappear.
+
+<img width="300px" src="https://i.giphy.com/media/3otPoFUhO2Z9uG1Qvm/giphy.webp"/>   
+
+*Navigation to tooltip, probably*
+
+One possiblity to do this is using events. The usual approach to this is to fire an event when the navigation component is being opened and then having the tooltip component listen to any events that might cause it to close. 
+
+### Challenge
+
+Copy the component you built in the previous step. When one of the buttons is clicked and the corresponding textbox is opened, the other textbox has to close. The rule is: Only one call-to-action textbox can be opened at all times.  
+
+### Hints and guidelines
+
++ As stated before, JS/ES6 is preffered. However, jQuery provides a very cool way of implementing events and adds functionality to them, that doesn't exist in plain JS. Try completing the challenge twice, once in plain JS and once in jQuery to see the difference
++ Optional challenge: Perhaps you are familiar with a more modern approach to component communication: State management. If you want to, you can complete the challenge using Redux or something similiar as well 
+
+### Resources
+
```

Comment:
https://medium.freecodecamp.org/learn-these-javascript-fundamentals-and-become-a-better-developer-2a031a0dc9cf
http://2ality.com/2015/08/getting-started-es6.html